packages/eas-cli/src/commands/build/cancel.ts

```diff
@@ -116,24 +89,24 @@ export async function selectBuildToCancelAsync(
   }
 }
 
-async function ensureBuildExistsAsync(
-  graphqlClient: ExpoGraphqlClient,
-  buildId: string
-): Promise<void> {
-  try {
-    await BuildQuery.byIdAsync(graphqlClient, buildId);
-  } catch {
-    throw new Error(`Couldn't find a build matching the id ${buildId}`);
-  }
-}
-
 export default class BuildCancel extends EasCommand {
   static override description = 'cancel a build';
 
   static override args = [{ name: 'BUILD_ID' }];
 
   static override flags = {
     ...EASNonInteractiveFlag,
+    platform: Flags.string({
+      char: 'p',
+      description: 'Platform for which to list builds when ID not provided',
+      options: ['android', 'ios', 'all'],
+    }),
+    profile: Flags.string({
+      char: 'e',
+      description:
+        'Name of the build profile from eas.json. Defaults to "production" if defined in eas.json.',
+      helpValue: 'PROFILE_NAME',
+    }),
```

Comment:
This is probably a nitpick, but can we throw an error if the combination of build ID and these flags is provided? I want to avoid any confusion for the users here. What I mean is a situation where users believe this flag will only delete a build with such id if it is of a given platform and profile.

Reply:
That's a valid point

Reply:
Updated